chef-config/lib/chef-config/config.rb

```diff
@@ -84,8 +84,10 @@ def self.platform_specific_path(path)
     # @return [String] the platform-specific path
     #
     def self.etc_chef_dir(windows: ChefUtils.windows?)
-      path = windows ? c_chef_dir : PathHelper.join("/etc", ChefUtils::Dist::Infra::DIR_SUFFIX, windows: windows)
-      PathHelper.cleanpath(path, windows: windows)
+      @etc_chef_dir ||= begin
```

Comment:
```suggestion
      @etc_chef_dir ||= {}
      @etc_chef_dir[windows] ||= begin
```

Kinda gross, but what about this?

Reply:
I'm not clear why that's better?

Reply:
This function takes a boolean that controls which of two values are returned, but this new caching mechanism will only cache the value for whichever param it is called with first.